README.md

```diff
@@ -4,221 +4,627 @@
+  For detailed information about platform condition expressions, see the [platform documentation](https://www.iguazio.com/docs/reference/latest-release/expressions/condition-expression/).
 
 Example:
 ```python
-data = [['tom', 10, 'TLV'], ['nick', 15, 'Berlin'], ['juli', 14, 'NY']] 
-df = pd.DataFrame(data, columns = ['name', 'age', 'city'])
-df.set_index('name')
-v3c.write(backend='kv', table='mytable', dfs=df, condition='age>14')
+data = [["tom", 10, "TLV"], ["nick", 15, "Berlin"], ["juli", 14, "NY"]]
+df = pd.DataFrame(data, columns = ["name", "age", "city"])
+df.set_index("name")
+v3c.write(backend="kv", table="mytable", dfs=df, condition="age>14")
 ```
 
-### Read
-Reads data from a backend.  
-Common read parameters:
-* iterator: bool - Return iterator of DataFrames or (if False) just one DataFrame
-* filter: string - Query filter (can't be used with query)
-* columns: []str - List of columns to pass (can't be used with query)
-* data_format: string - Data format (Not yet supported)
-* marker: string - Query marker (Not yet supported)
-* limit: int - Maximal number of rows to return (Not yet supported)
-* row_layout: bool - Weather to use row layout (vs the default column layout) (Not yet supported)
-
-
-#### backend specific parameters
-##### TSDB
-* start: string
-* end: string
-* step: string
-* aggregators: string
-* aggregationWindow: string
-* query: string - Query in SQL format
-* group_by: string - Query group by (can't be used with query)
-* multi_index: bool - Get the results as a multi index data frame where the labels are used as indexes
- in addition to the timestamp, or if `False` (default behavior) only the timestamp will function as the index.
-
-For detailed info on these parameters please visit [TSDB](https://github.com/v3io/v3io-tsdb#v3io-tsdb) docs.  
+<a id="method-read"></a>
+### read Method
+
+Reads data from a table or stream in a platform data container to a DataFrame, according to the configured backend.
+
+- [Syntax](#method-read-syntax)
+- [Common parameters](#method-read-common-params)
+- [`tsdb` backend `read` parameters](#method-read-params-tsdb)
+- [`kv` backend `read` parameters](#method-read-params-kv)
+- [`stream` backend `read` parameters](#method-read-params-stream)
+- [Return Value](#method-read-return-value)
+
+Reads data from a backend.
+
+<a id="method-read-syntax"></a>
+#### Syntax
+
+```python
+read(backend='', table='', query='', columns=None, filter='', group_by='',
+    limit=0, data_format='', row_layout=False, max_in_message=0, marker='',
+    iterator=False, **kw)
+```
+
+<a id="method-read-common-params"></a>
+#### Common read Parameters
+
+All Frames backends that support the `read` method support the following common parameters:
+
+- <a id="method-read-param-iterator"></a>**iterator** &mdash; (Optional) (default: `False`) &mdash; `bool` &mdash; `True` to return a DataFrames iterator; `False` to return a single DataFrame.
```

Comment:
better to order the parameters with the `optional` ones in the end

Reply:
@talIguaz I plan to reorder all parameter lists alphabetically and include separate requirement bullets (as I did in the constructor documentation). I just didn't have time to handle this for this PR + I want to do the reordering in a separate commit to simplify diffs.